queue_services/entity-emailer/src/entity_emailer/worker.py

```diff
@@ -74,6 +74,20 @@ async def publish_event(payload: dict):
 
 def send_email(email: dict, token: str):
     """Send the email."""
+    # stop processing email when payload is incompleted.
+    if not email \
+            or 'recipients' not in email \
+            or 'content' not in email \
+            or 'body' not in email['content']:
+        logger.debug('Send email: email object(s) is empty')
+        raise QueueException('Unsuccessful sending email.')
+
+    if not email['recipients'] \
+            or not email['content'] \
+            or not email['content']['body']:
+        logger.debug('Send email: email object(s) is missing')
+        raise QueueException('Unsuccessful sending email.')
```

Comment:
I think an ideal solution would be to throw another custom exception here and add another `except` block to `cb_subscription_handler`.  This`except` block would create or update a message in the tracker with a status of `TERMINATED` or something like that.  If we do that then OPs will be able to query tracker db for specific queue messages that were terminated intentionally by us.  With the current state of things, this means we would be able to pick out the NRs where we weren't able to retrieve a receipt from the pay api.

I am fine with this approach though of just throwing the `QueueException` to get the queue to stop re-processing the messages.  @thorwolpert does Patrick's current implementation work for you?